### PR TITLE
Add an option to disable the references CodeLens

### PR DESCRIPTION
**Pull request by rchande**
Fixes https://github.com/OmniSharp/omnisharp-vscode/issues/1570

### CODE REVIEW

**Comment by rchande** [comment]
@DustinCampbell I didn't find any tests for the references indicator. Can you confirm whether or not there are any?

**Comment by DustinCampbell** [comment]
There are no tests for the references indicator. You're are 100% welcome to explore writing one.

File: package.json [hunk 1]
```diff
@@ -373,6 +373,11 @@
           "type": "boolean",
           "default": true,
           "description": "Specifes whether OmniSharp should use VS Code editor settings for C# code formatting (use of tabs, indentation size)."
+        },
+        "omnisharp.showReferencesCodeLens": {
```
Comment (DustinCampbell) [inline package.json]: This should be "csharp.showReferencesCodeLens". The "omnisharp.*" prefixed settings are intended to be for OmniSharp itself (though we diverged in one or two cases).

File: src/features/codeLensProvider.ts [hunk 2]
```diff
@@ -43,6 +44,13 @@ export default class OmniSharpCodeLensProvider extends AbstractProvider implemen
     };
 
     provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken): vscode.CodeLens[] | Thenable<vscode.CodeLens[]> {
+        const options = Options.Read();
+        if (!options.showReferencesCodeLens)
+        {
+            let arr: vscode.CodeLens[] = [];
+            return arr;
```
Comment (DustinCampbell) [inline src/features/codeLensProvider.ts]: Can't use just return `[]` here? Or, does TypeScript complain.

File: package.json [hunk 3]
```diff
@@ -373,6 +373,11 @@
...
+          "type": "boolean",
+          "default": true,
+          "description": "Specifies whether Omnisharp should show the references CodeLens"
```
Comment (DustinCampbell) [inline package.json]: How about: "Specifies whether the references CodeLens should be shown".

Remember that the extension isn't "OmniSharp".

**Comment by rchande** [comment]
Thanks @DustinCampbell -- incorporated your feedback.
Regarding tests, I think maybe there's a larger workitem to go and add tests for all the features that depend on the OmniSharp server (would require some kind of mock of the service and so forth).

**Comment by DustinCampbell** [comment]
@rchande: I'm not sure a mock service is actually needed. VS Code extension tests run by launching an instance of VS Code and running the extension. If we could add the infrastructure to allow our platform specific components to download during test runs, I think it'd work fine.

**Comment by DustinCampbell** [comment]
Of course, that'd be more of an integration test. Adding a mock for the omnisharp service endpoints wouldn't be a big deal. Agree that it's a bigger work item.

**Comment by pnloyd** [comment]
@rchanda Thanks for this!

**Pull request merged**